packages/core/src/lib/telemetry.ts

```diff
@@ -0,0 +1,240 @@
+
+`;
+
+export function runTelemetry(
+  cwd: string,
+  lists: Record<string, InitialisedList>,
+  dbProviderName: DatabaseProvider
+) {
+  try {
+    const userConfig = getTelemetryConfig().userConfig;
+    const telemetry = getTelemetryConfig().telemetry;
+    if (
+      ci.isCI || // Don't run in CI
+      process.env.NODE_ENV === 'production' || // Don't run in production
+      telemetry === false // Don't run if the user has opted out
+    ) {
+      return;
+    }
+    if (telemetry === undefined) {
+      const newTelemetry: Configuration['telemetry'] = {
+        device: { informedAt: new Date().toISOString() },
+        projects: {
+          default: { informedAt: new Date().toISOString() },
+          [cwd]: { informedAt: new Date().toISOString() },
+        },
+      };
+      userConfig.set('telemetry', newTelemetry);
+      console.log(notifyText);
+      // Don't run telemetry on first run, give the user a chance to opt out
+      return;
+    }
+    if (!telemetry) {
+      return;
+    }
+    if (telemetry.projects[cwd] === undefined) {
+      userConfig.set(`telemetry.projects${cwd}`, telemetry.projects.default);
+      telemetry.projects[cwd] = telemetry.projects.default;
+    }
+    if (!!telemetry.projects[cwd]) {
+      sendProjectTelemetryEvent(cwd, lists, dbProviderName, telemetry.projects[cwd]);
+    }
+    if (!!telemetry.device) {
+      sendDeviceTelemetryEvent(telemetry.device);
+    }
+  } catch (err) {
+    // Fail silently unless KEYSTONE_TELEMETRY_DEBUG is set to '1'
+    if (process.env.KEYSTONE_TELEMETRY_DEBUG === '1') {
+      console.log(err);
+    }
+  }
+}
+
+const fieldCount = (lists?: Record<string, InitialisedList>): Project['fields'] => {
+  if (!lists) {
+    return { unknown: 0 };
+  }
+  const fields: Project['fields'] = { unknown: 0 };
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [_listKey, { resolvedDbFields }] of Object.entries(lists)) {
+    for (const [fieldPath, field] of Object.entries(resolvedDbFields)) {
+      const fieldType = field.__ksTelemetryFieldTypeName;
+
+      if (!fieldType) {
+        // skip id fields
+        if (fieldPath.endsWith('id')) continue;
+
+        //skip from relationship fields
+        if (fieldPath.startsWith('from')) continue;
+        if (field.kind === 'relation') {
+          fields['@keystone-6/relationship'] = (fields['@keystone-6/relationship'] || 0) + 1;
+          continue;
+        }
+        fields.unknown++;
+        continue;
+      }
+      if (!fields[fieldType]) {
+        fields[fieldType] = 0;
+      }
+      fields[fieldType] += 1;
+    }
+  }
+  return fields;
+};
+
+function sendEvent(eventType: 'project' | 'device', eventData: Project | Device) {
+  try {
+    const telemetryEndpoint = process.env.KEYSTONE_TELEMETRY_ENDPOINT || defaults.telemetryEndpoint;
+    const telemetryUrl = `${telemetryEndpoint}/v1/event/${eventType}`;
+
+    if (process.env.KEYSTONE_TELEMETRY_DISPLAY) {
```

Comment:
I think we should simply leave this to `keystone telemetry status`,  the number of environment variables is a bit unnecessary.

Additionally, if you want to still keep this for debugging purposes, we should use `KEYSTONE_TELEMETRY_DEBUG`